modules/home-environment.nix

```diff
@@ -598,13 +598,27 @@ in
         ''
       else
         ''
+          function nixProfileList() {
+            # We attempt to use `--json` first (added in Nix 2.17). Otherwise attempt to
+            # parse the legacy output format.
+            {
+              nix profile list --json 2>/dev/null \
+                | jq -r ".elements[].storePaths[] | select(endswith(\"$1\"))"
+            } || {
+              nix profile list \
+                | { grep "$1\$" || test $? = 1; } \
+                | cut -d ' ' -f 4
+            }
+          }
+
+          function nixRemoveProfileByName() {
+              nixProfileList "$1" | xargs -t $DRY_RUN_CMD nix profile remove $VERBOSE_ARG
+          }
+
```

Comment:
This should probably be deduplicated?

Reply:
You mean deduplicated between here and the home-manager tool? I would agree, the idea was that this would be a temporary solution until https://github.com/NixOS/nix/issues/6349 was resolved but I fear it may be a while. Will have to think about how to resolve it better in HM.